runtime/compiler/ilgen/IlGenerator.cpp

```diff
@@ -152,8 +152,7 @@ TR_J9ByteCodeIlGenerator::genIL()
           * if DelayRelocationForAOT don't persist iprofiler info now.
           * instead, persist iprofiler info when loading the aot compilation
           */
-         TR_J9SharedCacheVM *fej9sc = (TR_J9SharedCacheVM *)(comp()->fej9());
-         if(comp()->getOption(TR_DisableDelayRelocationForAOTCompilations) || !fej9sc->shouldDelayAotLoad())
+         if (comp()->getOption(TR_DisableDelayRelocationForAOTCompilations) || !comp()->fej9()->shouldDelayAotLoad())
```

Comment:
https://github.com/eclipse/openj9/pull/4599#discussion_r254037452

Reply:
^ the origin on this decision. So apparently for AOT `fej9()->shouldDelayAotLoad()` is supposed to return true. It's the case for TR_J9SharedCacheVM, so it makes sense for TR_J9SharedCacheServerVM to return true since it's only used for AOT.